### PR TITLE
with logits_all == true, use to the last logits vector

### PR DESCRIPTION
**Pull request by Piezoid**
Affects `llama_sample_top_p_top_k` and `params.ignore_eos` mode in `main.cpp`.
Alternatively the sampler could take an additional argument indicating the position to sample in the batch.
As of now, I haven't come across any code path that hit this issue (beside my changes in #477).

### CODE REVIEW

**Comment by Piezoid** [comment]
Resolved in 03f7e335.
